example/log-label-example/README.md

```diff
@@ -0,0 +1,97 @@
+# Custom Log Label Examples
+
+Examples illustrating how to add custom labels to logs generated by `userale`.
+
+## Adding Custom Labels to Logs
+
+### Method 1
+
+Consider the following HTML:
+
+```html
+
+<div>
+    <button>New Feature</button>
+</div>
+```
+
+The following code snippet will add a custom field and send a log whenever the new feature button is clicked:
+
+```js
+document.addEventListener('click', function (e) {
+    if (e.target.innerHTML === 'New Feature') {
+        window.userale.map(log => ({...log, logType: 'custom', customLabel: 'New Feature'}));
+        window.userale.packageLog(e, window.userale.details(window.userale.options(), e.type));
+        window.userale.map();
+    } else {
+        return false
+    }
+});
```

Comment:
this works, but i think it relies a bit on the implementation details of `map`/`packageLog`. the intended usage is captured in the (non-highlighted) example in the main `README.md`.

when we call the `userale.map` function, it keeps the function you pass it around until it gets re-assigned. the given function gets called for _every_ log that the library receives. with that in mind, there is no need to setup your own event listener, as that is what userale is for. all you need to 

here is how i'd imagine the above code would be implemented:
```js
// no need to set up a separate listener
window.userale.map((log, e) => {
    // determine whether we want to modify the log
    if (e.target.innerHTML !== 'New Feature') {
        return log; // return the log if we don't care about this event
    }

    // return the log with any changes we want
    return {
        ...log,
        customLabel: 'New Feature',
        logType: 'custom',
    };
});
```

Reply:
if you want to make the code extra compact (not something i am a huge fan of), you can write it like this:
```js
window.userale.map((log, e) => e.target.innerHTML === 'New Feature'
    ? { ...log, customLabel: 'New Feature', logType: 'custom' }
    : log);
```

Reply:
Agreed Rob, I also prefer the idea of just using the `userale.map` function as opposed to adding more event listeners. I think we need a bit more context on what exactly the users are trying to accomplish and what they want to [decorate](https://github.com/apache/incubator-flagon-useralejs/issues/63).

The examples that Rob linked and are located at [example/index.js](https://github.com/apache/incubator-flagon-useralejs/blob/master/example/index.js). Do you have any idea why in that code the extra event listeners are added?

Reply:
+1  For labeling, we should not advocate that users create new event listeners. Ideally, users should only be creating new events listeners if (1) the events are not captured by UserALE.js natively, or (2) they want to filter out everything but custom logs.

Reply:
> if you want to make the code extra compact (not something i am a huge fan of), you can write it like this:
> 
> ```js
> window.userale.map((log, e) => e.type ==='click' && e.target.innerHTML === 'New Feature'
>     ? { ...log, customLabel: 'New Feature', logType: 'custom' }
>     : log);
> ```

+1 agree with Rob. Worth being verbose in examples for accessibility and descriptiveness. If they want to compact our examples, that's not hard.